src/get_maintainer_wrapper.sh

```diff
@@ -1,90 +1,240 @@
-    #
-    # MODULE_AUTHOR ("a_long_email@xx.com" \
-    #                "another_long@yy.com" )
-    # and:
-    #
-    # MODULE_AUTHOR ("a_long_email@xx.com"
-    #                "another_long@yy.com" )
-    #
-    # which are, both, valid C statements. In the currently
-    # implementation neither of these four emails will be printed.
-
-    local FILE_OR_DIR=$1
-    local files=( )
-    if [[ -d $FILE_OR_DIR ]]; then
-        for file in $FILE_OR_DIR/*; do
-            if [[ -f $file ]]; then
-                files+=($file)
-            fi
-        done
-    elif [[ -f $FILE_OR_DIR ]]; then
-        files+=($FILE_OR_DIR)
+  # TODO: currently only authors found in a single line MODULE_AUTHOR
+  # statement are captured and printed. In the future, it would be
+  # nice to look for multiline MODULE_AUTHOR statatements such as:
+  #
+  # MODULE_AUTHOR ("a_long_email@xx.com" \
+  #                "another_long@yy.com" )
+  # and:
+  #
+  # MODULE_AUTHOR ("a_long_email@xx.com"
+  #                "another_long@yy.com" )
+  #
+  # which are, both, valid C statements. In the currently
+  # implementation neither of these four emails will be printed.
+
+  local FILE_OR_DIR=$1
+  local files=( )
+  if [[ -d $FILE_OR_DIR ]]; then
+    for file in $FILE_OR_DIR/*; do
+      if [[ -f $file ]]; then
+        files+=($file)
+      fi
+    done
+  elif [[ -f $FILE_OR_DIR ]]; then
+    files+=($FILE_OR_DIR)
+  fi
+
+  local printed_authors_separator=false
+
+  for file in ${files[@]}; do
+    authors=$(grep -oE "MODULE_AUTHOR *\(.*\)" $file |
+              sed -E "s/(MODULE_AUTHOR *\( *\"|\" *\))//g" |
+              sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/, /g' )
+    if [[ ! -z $authors ]]; then
+      if [[ $printed_authors_separator = false ]]; then
+        say $SEPARATOR
+        say "MODULE AUTHORS:"
+        printed_authors_separator=true
+      fi
+      say -n "$(basename $file): "
+      echo "$authors"
     fi
+  done
+}
+
+# Checks if a directory is a kernel tree root
+#
+# @parameter A directory path
+#
+# Returns:
+# True if given dir is a kernel tree root and false otherwise.
+function is_kernel_root()
+{
+  local -r DIR="$@"
+  # The following files are some of the files expected to be at a linux
+  # tree root and not expected to change. Their presence (or abscense)
+  # is used to tell if a directory is a linux tree root or not. (They
+  # are the same ones used by get_maintainer.pl)
+  if [[ -f "${DIR}/COPYING" &&
+        -f "${DIR}/CREDITS" &&
+        -f "${DIR}/Kbuild" &&
+        -e "${DIR}/MAINTAINERS" &&
+        -f "${DIR}/Makefile" &&
+        -f "${DIR}/README" &&
+        -d "${DIR}/Documentation" &&
+        -d "${DIR}/arch" &&
+        -d "${DIR}/include" &&
+        -d "${DIR}/drivers" &&
+        -d "${DIR}/fs" &&
+        -d "${DIR}/init" &&
+        -d "${DIR}/ipc" &&
+        -d "${DIR}/kernel" &&
+        -d "${DIR}/lib" &&
+        -d "${DIR}/scripts" ]]; then
```

Comment:
I’m a little bit worried about this part since it makes the “kw m” fragile. If the Linux Kernel changes a single file, “kw m” will fail. Also, this hardcoded option could be a little bit more flexible.

My suggestions:
1. Add three or four files for checking if the directory is a kernel root or not. I suggest you inspect for get_maintainers.pl, MAINTAINERS, codestyle.pl, and kernel directory.
2. Add a second if, that loads an extra variable from etc/kworkflow.config. This variable could have other files specified by the user; with this, if ‘kw m’ fails to match the root kernel, the user can add a fine-grained level for an extra check.